src/robusta/core/sinks/opsgenie/opsgenie_sink.py

```diff
@@ -73,17 +83,48 @@ def __ack_alert(self, fingerprint: str, user: str, note: str):
         except opsgenie_sdk.ApiException as err:
             logging.error(f"Error acking opsGenie alert {fingerprint} {err}", exc_info=True)
 
+    def __get_teams(self, finding: Finding) -> List[str]:
+        teams = []
+        for team_template in self.teams:
+            try:
+                if "$" in team_template:
+                    # Only process templates that contain variables
+                    team = ChannelTransformer.template(
+                        team_template,
+                        self.default_team,  # Use default_team as fallback
+                        self.cluster_name,
+                        finding.subject.labels,
+                        finding.subject.annotations,
+                    )
+                    if team:  # Only add non-null teams
+                        teams.append(team)
+                else:
+                    # Use static team name directly
+                    teams.append(team_template)
+            except Exception as e:
+                logging.warning(
+                    f"Failed to process team template {team_template} for alert subject {finding.service_key}: {e}"
+                )
+                continue
+        return teams
+
     def __open_alert(self, finding: Finding, platform_enabled: bool):
         description = self.__to_description(finding, platform_enabled)
         details = self.__to_details(finding)
         tags = self.tags.copy()
         tags.insert(0, self.cluster_name)
+
+        # Get teams based on templates
+        teams = self.__get_teams(finding)
+        if not teams and self.teams:  # dynamic routing failed and no default team configured
+            logging.warning(f"No valid teams resolved for finding {finding.title}. Alert may not be routed properly.")
+
         body = opsgenie_sdk.CreateAlertPayload(
             source="Robusta",
             message=finding.title,
             description=description,
             alias=finding.fingerprint,
-            responders=[{"name": team, "type": "team"} for team in self.teams],
+            responders=[{"name": team, "type": "team"} for team in teams],
```

Comment:
Just to make sure, if teams is [] the responsers will be [] as well. Is that valid value for responders field? I looked on the docs, they aren't say it is explicitly so just make sure it won't cause error.

Reply:
no issue